mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -767,14 +767,27 @@ class WidgetAdapter(
             } else {
                 TimeFormat.CLOCK_12H
             }
+            val prefs = itemView.context.getPrefs()
+            val inputMode = prefs.getInt(PrefKeys.TIME_PICKER_INPUT_MODE, MaterialTimePicker.INPUT_MODE_CLOCK)
             val timePicker = MaterialTimePicker.Builder()
                 .setTimeFormat(timeFormat)
                 .setHour(date.hour)
                 .setMinute(date.minute)
+                .setInputMode(inputMode)
                 .build()
+
+            fun saveInputType() {
+                prefs.edit {
+                    putInt(PrefKeys.TIME_PICKER_INPUT_MODE, timePicker.inputMode)
+                }
+            }
+
             timePicker.addOnPositiveButtonClickListener {
                 sendUpdate(widget, date.withHour(timePicker.hour).withMinute(timePicker.minute))
+                saveInputType()
             }
+            timePicker.addOnCancelListener { saveInputType() }
```

Comment:
Cancellation should cause the dismiss listener to be called, thus the cancel listener should not be necessary.

Reply:
DismissListener is also called when the "OK" button is pressed. 👍